XCUITests/NewTabSettings.swift

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+class NewTabSettingsTest: BaseTestCase {
+    func testCheckNewTabSettingsByDefault() {
+        navigator.goto(NewTabSettings)
+        waitforExistence(app.navigationBars["New Tab Settings"])
+        waitforExistence(app.tables.cells.staticTexts["Show your Top Sites"])
+        XCTAssertTrue(app.tables.switches["ASPocketStoriesVisible"].isEnabled)
+        XCTAssertTrue(app.tables.switches["ASBookmarkHighlightsVisible"].isEnabled)
+        XCTAssertTrue(app.tables.switches["ASRecentHighlightsVisible"].isEnabled)
+    }
+
+    func testToggleOffOnAdditionalContentBookmarks() {
+        // Bookmark one site and check it appears in a new tab
+        navigator.performAction(Action.BookmarkThreeDots)
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        waitforExistence(app.staticTexts["Highlights"])
+
+        // Disable toggle and check that it does not appear in a new tab
+        navigator.goto(NewTabSettings)
+        navigator.toggleOff(userState.bookmarksInNewTab, withAction: Action.ToggleBookmarksInNewTab)
+        navigator.goto(HomePanelsScreen)
+        navigator.goto(NewTabScreen)
+
+        //This appears under top sites
+        waitforNoExistence(app.staticTexts["Highlights"])
+
+        // Enable toggle again and check it is shown
+        navigator.goto(NewTabSettings)
+        navigator.toggleOn(userState.bookmarksInNewTab, withAction: Action.ToggleBookmarksInNewTab)
+        navigator.goto(HomePanelsScreen)
+        navigator.goto(NewTabScreen)
+        navigator.nowAt(NewTabScreen)
+        waitforExistence(app.staticTexts["Highlights"])
+    }
+
+    func testChangeNewTabSettingsShowBlankPage() {
+        navigator.goto(NewTabChoiceSettings)
+        waitforExistence(app.tables["NewTabPage.Setting.Options"])
+
+        navigator.performAction(Action.ShowBlankPage)
+        navigator.goto(HomePanelsScreen)
+        navigator.goto(NewTabScreen)
+
+        waitforNoExistence(app.collectionViews.cells["TopSitesCell"])
+        waitforNoExistence(app.collectionViews.cells["TopSitesCell"].collectionViews.cells["youtube"])
+        waitforNoExistence(app.staticTexts["Highlights"])
+    }
+
+    func testChangeNewTabSettingsShowYourBookmarks() {
+        navigator.goto(NewTabChoiceSettings)
+        waitforExistence(app.tables["NewTabPage.Setting.Options"])
+        // Show Bookmarks panel without bookmarks
+        navigator.performAction(Action.ShowBookmarksPage)
+        navigator.goto(HomePanelsScreen)
+        navigator.goto(NewTabScreen)
```

Comment:
You do this several times in this test: goto `HomePanelsScreen` then `NewTabScreen`. Why?

Reply:
I needed to open a new tab to check that the setting selected was working that's whay I did it that way. Just realized I can change those two lines with: ```navigator.performAction(Action.OpenNewTabFromTabTray)```
I will update the tests.
Thanks